src/hash.rs

```diff
@@ -64,3 +74,51 @@ fn file_hash(algorithm: &str, path: &str) -> Result<String> {
 
     Ok(hash_algorithm(algorithm, &bytes)?)
 }
+
+fn totp_generate_tolerance(hex_seed: &str, tolerance: i32) -> Result<String> {
+    let mut result: String = "".to_string();
+    for i in -tolerance..(tolerance + 1) {
+        if i != -tolerance {
+            result.push(',');
```

Comment:
Perhaps should output valid JSON instead. I think we have some other functions that do this?